Gordon360/Services/RecIM/ParticipantService.cs

```diff
@@ -136,21 +144,116 @@ public IEnumerable<TeamExtendedViewModel> GetParticipantTeams(string username)
 
         public IEnumerable<ParticipantExtendedViewModel> GetParticipants()
         {
-            //temporary, slow and will be adjusted after we implement views in the DB
-            var participants = _context.Participant.Select(p => new ParticipantExtendedViewModel
-            {
-                Username = p.Username,
-                Status = p.ParticipantStatusHistory.OrderByDescending(psh => psh.ID).FirstOrDefault().Status.Description,
-                IsAdmin = p.IsAdmin
-            });
-            return participants;
+            // This is Participant left join Student left join FacStaff to get each participant's firstname and lastname
```

Comment:
All of this custom participant logic would be much cleaner and more usable as view of Participants. You could have a table of non-custom participants which stores just the RecIM specific data, as well as custom participant data table that stores the custom data for external participants, and then a view that displays all the participants, taking their data from the Student/FacStaff view or the Custom Participant table, as applicable.

Reply:
@Kevin-Chu58